frontend/utils/region.ts

```diff
@@ -1,81 +1,111 @@
+  ["asia-southeast2", "Asia Pacific (Jakarta)", "asia-pacific-jakarta"],
+  ["australia-southeast1", "Asia Pacific (Sydney)", "asia-pacific-sydney"],
+  [
+    "australia-southeast2",
+    "Asia Pacific (Melbourne)",
+    "asia-pacific-melbourne",
+  ],
+  ["europe-central2", "Europe (Warsaw)", "europe-warsaw"],
+  ["europe-north1", "Europe (Finland)", "europe-finland"],
+  ["europe-west1", "Europe (Belgium)", "europe-belgium"],
+  ["europe-west2", "Europe (London)", "europe-london"],
+  ["europe-west3", "Europe (Frankfurt)", "europe-frankfurt"],
+  ["europe-west4", "Europe (Netherlands)", "europe-netherlands"],
+  ["europe-west6", "Europe (Zurich)", "europe-zurich"],
+  ["europe-west8", "Europe (Milan)", "europe-milan"],
+  ["europe-west9", "Europe (Paris)", "europe-paris"],
+  ["northamerica-northeast1", "Canada (Montréal)", "canada-montreal"],
+  ["northamerica-northeast2", "Canada (Toronto)", "canada-toronto"],
+  ["southamerica-east1", "South America (Osasco)", "south-america-osasco"],
+  ["southamerica-west1", "South America (Santiago)", "south-america-santiago"],
+  ["us-central1", "US Central (Iowa)", "us-central-iowa"],
+  ["us-east1", "US East (South Carolina)", "us-east-south-carolina"],
+  ["us-east4", "US East (N. Virginia)", "us-east-n-virginia"],
+  ["us-west1", "US West (Oregon)", "us-west-oregon"],
+  ["us-west2", "US West (Los Angeles)", "us-west-los-angeles"],
+  ["us-west3", "US West (Salt Lake City)", "us-west-salt-lake-city"],
+  ["us-west4", "US West (Las Vegas)", "us-west-las-vegas"],
+];
 
 export const getRegionName = (regionCode: string): string => {
-  if (!regionCodeMap.has(regionCode)) {
+  if (!regionCodeNameSlugMap.some(([code]) => code === regionCode)) {
     // If the code is not included in the map, we just show user the code.
     // Also, we need to update the map manually.
-    regionCodeMap.set(regionCode, `Other (${regionCode})`);
+    regionCodeNameSlugMap.push([
+      regionCode,
+      `Other (${regionCode})`,
+      slugify(`Other (${regionCode})`),
+    ]);
   }
-  return regionCodeMap.get(regionCode) as string;
+  return regionCodeNameSlugMap.find(([code]) => code === regionCode)![1];
 };
 
 export const getRegionCode = (regionName: string): string[] => {
   const regionCode: string[] = [];
-  regionCodeMap.forEach((name, code) => {
+  regionCodeNameSlugMap.forEach(([code, name]) => {
     if (name === regionName) {
       regionCode.push(code);
     }
   });
 
   return regionCode;
 };
+
+// "Asia Pacific (Hong Kong)" -> "Asia Pacific"
+export const getRegionPrefix = (regionName: string): string =>
+  regionName.substring(0, regionName.indexOf("(")).trimEnd();
+
+export const getRegionListByPrefix = (prefix: string): string[] =>
+  Array.from(
+    new Set(
+      regionCodeNameSlugMap
+        .filter((pair) => pair[1].startsWith(prefix))
+        .map((pair) => pair[1])
+    )
+  );
+
+// "US East (N. Virginia)" -> "us-east-n-virginia"
+export const slugify = (regionName: string): string =>
```

Comment:
Not a safe implementation since we don't know what special chars will be in the regionName.
Better to use a lib such as https://www.npmjs.com/package/slug